js/client-chat.js

```diff
@@ -93,7 +93,8 @@
 			}
 			this.tabComplete.reset();
 			this.chatHistory.push(text);
-			text = this.parseCommand(text);
+			// make sure commands in multiline inputs are handled correctly
+			text = text.split('\n').map(this.parseCommand, this).filter(function (line) {return (line !== false);}).join('\n');
```

Comment:
client commands + text is always at least a bit confusing of a situation.

My proposed change would make it slightly less confusing, overall.